src/main/java/hybrid/it/internship/library/service/impl/RentServiceImpl.java

```diff
@@ -21,4 +42,89 @@ public boolean existsByBookId(Long bookId) {
     public boolean existsByUserId(Long userId) {
         return rentRepository.existsByUserId(userId);
     }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<RentDTO> getOverdueRents() {
+
+        List<RentDTO> rentsDTOs = rentMapper.toDTO(rentRepository.findAll());
+
+        rentsDTOs.removeIf(rent -> !rent.getRentDate().plusDays(rentPeriodInDays).isBefore(LocalDateTime.now()));
+
+        return rentsDTOs;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<RentedBookDTO> getMostRented() {
+
```

Comment:
Generally avoid having methods this big. They are hard to read and make changes. Each method should have only one responsibility. In this case the method should splitted into a couple of private methods, for instance fetch(), sort() and map()